### DISCUSSION TITLE
Sort favorites

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#92] [Q&A] [answered]
One thing I don't like is that the option to sort the "favorites" by distance has been lost, sometimes you don't know the name you put in one of them and it's good to see what you have nearby.

![imatge](https://user-images.githubusercontent.com/122691401/226657679-b4d42610-3a77-4614-bb71-09d810ea93e5.png)

_Originally posted by @ramondevic in https://github.com/devemux86/cruiser/discussions/74#discussioncomment-5382806_

**Comment by devemux86** [comment]
> the option to sort the "favorites" by distance has been lost

Thanks for noticing and asking!

Sort is currently "hidden" because of the work in [favorite categories](https://github.com/devemux86/cruiser/discussions/45) and to discuss how you would like it to work.

_Favorites are automatically sorted by name_.
Do we need to sort them in different ways?

**Reply by ramondevic** [reply]
I like the old option chosing the order for name or distance

**Reply by devemux86** [reply]
There is no problem, we can enable again the sort options! :slightly_smiling_face: 

With the [favorite categories](https://github.com/devemux86/cruiser/discussions/45), everything becomes more complicated,
so we have to see how the workflow can remain simple for the users.

**Comment by devemux86** [comment]
@ramondevic and everyone

Sort favorites button can return in the next Cruiser update.

Do we also need the up / down :arrow_up: :arrow_down: buttons?
Sort by name or distance or ... should be ok?

With the new features coming in favorites (categories, colors, etc.)
their toolbar may become too crowded to use easily. :thinking:

**Reply by WolfgangZ42** [reply]
> Do we also need the up / down ⬆️ ⬇️ buttons?

In the past I used these buttons to sort by usage of the favorites - but it's really not importent, sort by name is okay for me.

**Reply by ramondevic** [reply]
For me, just name and distance :)

**Comment by devemux86** [comment, marked as answer]
Sort favorites is available again in [Cruiser 3.0.14](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

@ramondevic thanks for the report!

**Comment by MotoUKRider** [comment]
A minor "thing" that I've noticed...

When sorting Favorites by "Distance", then moving the map to a new location, you have to press the "sort" button again to get the Favorites nearby the current map area.  Should the Favorites list (when sorting via Distance) invoke a refresh based-upon the map position?

**Reply by devemux86** [reply]
@MotoUKRider 

Do you mean when the favorites list appears,
it should remember the last sort selection
and sort again the favorites (by name or distance)?

Currently the sort selection is not saved between sessions, but we can change it.
I am [rewriting](https://github.com/devemux86/cruiser/discussions/45) the favorites, so we can improve and change anything you like. :slightly_smiling_face:

**Reply by MotoUKRider** [reply]
Yes, when you invoke the list again after moving the map the old distances remain - requiring you to press the sort button to recalculate the new distance list.

**Reply by devemux86** [reply]
@MotoUKRider thanks for the idea!

I have put it in my list to see how we can provide it.

**Reply by MotoUKRider** [reply]
Am I imagining things or was this working in a previous build (or 2) because it doesn't seem to anymore?

When GPS-offline, I move the map and the call-up the Favorites Group but the old distances remain.  I have to long-press an entry and then invoke the Name/Distance Sort option.

**Reply by devemux86** [reply]
@MotoUKRider the sort does change automatically.

The distances are automatically calculated from:
- User location (if exists)
- Or map center

**Reply by devemux86** [reply]
@MotoUKRider I tested it again and it seems to work properly.

Are you sure the GPS location marker is not somewhere on the map?
If GPS location is outside the screen, it shows an indicator at screen border.

Even if close GPS sensor, the GPS location may be cached for some time.

**Reply by MotoUKRider** [reply]
No, it's not switched-on - I killed the app in the settings, too.
Moving the map and looking at Favorites in that Group does not update the list in terms of Distance. It remains on the last "sort" I did.

I now have a few Groups, with only the BaseCamp export (of my places) visible.  I'm sure it used to work just fine, bizarre!

**Reply by devemux86** [reply]
@MotoUKRider 

Do you mean the order of the favorites?
The sort order does not change automatically.

Or the blue distances under each favorite?
These are updated automatically each time you open the list.

**Reply by MotoUKRider** [reply]
Ah - I see what you mean.  The distances *do* indeed change, but the order doesn't, that's what confuses me.  I had expected the list to re-populate in the order I had defined.

**Reply by devemux86** [reply]
@MotoUKRider currently the sort order is not automatic, we should call it when we need it.

We could offer an automatic sort with name or distance if there is interest.

**Comment by ramondevic** [comment]
I would prefer that the sorting was not automatic but that the user had to click to sort, automations help but sometimes they confuse you.

**Reply by devemux86** [reply]
We could probably have a setting for sort favorites (manual vs auto)?
- Manual
- Name
- Distance

**Reply by MotoUKRider** [reply]
I just checked the BaseCamp GPX file that I use to import mine from, and there's a time-stamp within the `<wpt>` tag: `<time>2023-04-02T17:11:32Z</time>`

So if you have the provision to time-stamp manually added ones within Cruiser, there is the possibility of allowing those who import from other places via GPX to have a similar "experience" :)

**Comment by devemux86** [comment]
I suppose you want to be able to set the favorites in custom order?

**Reply by WolfgangZ42** [reply]
As I wrote 3 weeks ago, would bei nice ... but not important.
